### PR TITLE
Adds a new rock-paper-scissors emote

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Adds a new rock-paper-scissors emote, which lets you engage in a quick RPS game with someone standing nearby. It functions somewhat similar to daps and handshakes, if you know how those work.

You can just use `*rps`, which will pop up a little radial menu letting you choose between rock, paper, or scissors. Also, just like with handshakes, you can click someone who's readied a move with help intent to instantly get ready to play with them.
Otherwise, if you're feeling spicy, you can use emote parameters (remember we have those? I thought they were cool). Just emote `*rps-r`, or `*rps-rock` (same applying for [s]cissors and [p]aper) to quickly ready your move.

## Why It's Good For The Game
Being able to play rock paper scissors whenever you want is pretty sick. Either just to blow off steam, or to decide what to do.
I unironically think this may improve command's decision making.

## Images of changes
image
image
image

## Testing
Played rock paper scissors against myself many times, through both emotes and the radial.

## Changelog
:cl:
add: Adds an emote to play rock-paper-scissors, accessible with *rps.
/:cl:

### CODE REVIEW

**Comment by SynthTwo** [comment]
mrps (Medium Rock Paper Scissors)

File: code/datums/status_effects/neutral.dm [hunk 1]
```diff
@@ -144,6 +156,96 @@
...
+		if(RPS_EMOTE_TIE)
+			outcome_msg = "It's a tie!"
+		if(RPS_EMOTE_WE_WIN)
+			outcome_msg = "[user] wins!"
+		if(RPS_EMOTE_THEY_WIN)
+			outcome_msg = "[highfived] wins!"
+
+	user.visible_message(
+		"<span class='notice'>[user] plays <b>[move]</b>, and [highfived] plays <b>[their_status_effect.move]</b>.</span>",
+		"<span class='notice'>[highfived] plays <b>[their_status_effect.move]</b></span>.",
```
Comment (Henri215) [inline code/datums/status_effects/neutral.dm:210]:
```suggestion
		"<span class='notice'>[highfived] plays <b>[their_status_effect.move]</b>.</span>",
```

File: code/datums/status_effects/neutral.dm [hunk 2]
```diff
@@ -144,6 +156,96 @@
...
+/datum/status_effect/high_five/rps/on_apply()
+	if(!isnull(move))
+		to_chat(owner, "<span class='notice'>You prepare to play <b>[move]</b>.")
+		return ..()  // we already have the move, probably from the emote passing it in
+
+	move = get_rock_paper_scissors_move(owner)
+	if(move == null)
+		return FALSE  // make it auto-remove itself
+
+	to_chat(owner, "<span class='notice'>You prepare to play <b>[move]</b>.")
```
Comment (Henri215) [inline code/datums/status_effects/neutral.dm:237]:
```suggestion
		to_chat(owner, "<span class='notice'>You prepare to play <b>[move]</b>.</span>")
		return ..()  // we already have the move, probably from the emote passing it in

	move = get_rock_paper_scissors_move(owner)
	if(move == null)
		return FALSE  // make it auto-remove itself

	to_chat(owner, "<span class='notice'>You prepare to play <b>[move]</b>.</span>")
```

File: code/__DEFINES/emotes_defines.dm [hunk 3]
```diff
@@ -75,3 +77,12 @@
...
 #define USABLE_DEAD_EMOTES list("*flip", "*spin")
+
+// Strings used for the rock paper scissors emote and status effect
+#define RPS_EMOTE_ROCK 		"rock"
+#define RPS_EMOTE_PAPER 	"paper"
+#define RPS_EMOTE_SCISSORS 	"scissors"
+
+#define RPS_EMOTE_THEY_WIN		"aww"
+#define RPS_EMOTE_WE_WIN		"yay"
+#define RPS_EMOTE_TIE			"tie"
```
Comment (1080pCat) [inline code/__DEFINES/emotes_defines.dm:88]: github will vaporize the indents if i try to edit them, but can you try to pull the quotes over a tad, they look out of place.
Also what is with those names, i hope the quoted parts aren't use anywhere? im not awfully sure what the second part of defines are for.

File: code/datums/status_effects/neutral.dm [hunk 4]
```diff
@@ -69,6 +69,20 @@
 	user.remove_status_effect(type)
 	highfived.remove_status_effect(type)
 
+/datum/status_effect/high_five/proc/wiz_effect(mob/living/carbon/user, mob/living/carbon/highfived)
+	user.status_flags |= GODMODE
+	highfived.status_flags |= GODMODE
+	explosion(get_turf(user), 5, 2, 1, 3, cause = id)
+	// explosions have a spawn so this makes sure that we don't get gibbed
+	addtimer(CALLBACK(src, PROC_REF(wiz_cleanup), user, highfived), 0.3 SECONDS) //I want to be sure this lasts long enough, with lag.
```
Comment (1080pCat) [inline code/datums/status_effects/neutral.dm:77]:
```suggestion
	addtimer(CALLBACK(src, PROC_REF(wiz_cleanup), user, highfived), 0.3 SECONDS) // I want to be sure this lasts long enough, with lag.
```

File: code/__DEFINES/emotes_defines.dm (hunk 3 above)
Comment (lewcc) [reply code/__DEFINES/emotes_defines.dm:88]: Nah, the second part of the defines is pretty much arbitrary, as long as they're unique here. I just figured I'd get silly with it.

I can probably move the tabs over tho

**Review by Burzah — APPROVED** [review]
Halfway through I had to stop and check to make sure I wasn't looking at high five code. 😄  Looks good!

**Pull request merged**